src/python/WMCore/WMBS/Oracle/Locations/New.py

```diff
@@ -17,10 +17,14 @@ class New(NewLocationMySQL):
              WHERE NOT EXISTS (SELECT site_name FROM wmbs_location
                                WHERE site_name = :location)"""
 
-    seSQL = """INSERT INTO wmbs_location_pnns (location, pnn)
-                 SELECT wl.id, :pnn FROM wmbs_location wl
-                 WHERE wl.site_name = :location
-                 AND NOT EXISTS (SELECT null FROM wmbs_location_pnns wls2 WHERE
-                                  wls2.pnn = :pnn
-                                  AND wls2.location = wl.id)
-                 """
+    pnnSQL = """INSERT INTO wmbs_pnns (id, pnn)
+                  SELECT wmbs_pnns_SEQ.nextval, :pnn
+                  FROM DUAL WHERE NOT EXISTS
+                (SELECT pnn FROM wmbs_pnns where pnn = :pnn)"""
+
+    mapSQL = """INSERT INTO wmbs_location_pnns (location, pnn)
```

Comment:
Damn you Oracle!!!
@ticoann are you able to spot anything wrong with this query? Oracle gives me a general error message like:
```
DatabaseError: (DatabaseError) ORA-00933: SQL command not properly ended
 'INSERT INTO wmbs_location_pnns (location, pnn)\n                  SELECT wl.id, wpnn.id FROM wmbs_location wl, wmbs_pnns wpnn\n                  WHERE wl.site_name = :location AND wpnn.pnn = :pnn\n                FROM DUAL WHERE NOT EXISTS\n                (SELECT wmbs_location.site_name, wmbs_pnns.pnn FROM wmbs_location, wmbs_pnns\n                WHERE wmbs_location.site_name = :location AND wmbs_pnns.pnn = :pnn)' {'location': 'malpaquet', 'pnn': 'T2_CH_CERN'}
-------------------- >> begin captured logging << --------------------
```
when running ErrorHandler_t.py:ErrorHandlerTest.testE_FailJobs

Reply:
Compare to previous query it seems it is using  just AND NOT EXISTS without DUAL. But I am not sure why.
Try following query.

```
mapSQL = """INSERT INTO wmbs_location_pnns (location, pnn)
               SELECT wl.id, wpnn.id FROM wmbs_location wl, wmbs_pnns wpnn
                 WHERE wl.site_name = :location AND wpnn.pnn = :pnn
             AND NOT EXISTS  (SELECT null FROM wmbs_location wl, wmbs_pnns wpnn
              WHERE wl.site_name = :location AND wpnn.pnn = :pnn)"""
```

Reply:
You're my hero! This one works!

Reply:
I wonder whether there is any reason not to use always that syntax (select null etc etc) every time we need to `insert ignore` into oracle.... would you know Seangchan?

Reply:
I am not sure. I am just master of copy and paste. :smile: I think DUAL table is used when oracle need values but not related to specific table.  I am not sure how we are using DUAL table in general but I don't think we have to use every INSERT IGNORE case.